src/event-engine/states/handshake_stopped.ts

```diff
@@ -1,20 +1,45 @@
+import { Cursor } from '../../models/Cursor';
 import { State } from '../core/state';
 import { Effects } from '../effects';
-import { Events, reconnect, subscriptionChange } from '../events';
+import { Events, reconnect, restore, subscriptionChange, unsubscribeAll } from '../events';
 import { HandshakingState } from './handshaking';
+import { UnsubscribedState } from './unsubscribed';
 
 type HandshakeStoppedStateContext = {
   channels: string[];
   groups: string[];
+  cursor?: Cursor;
 };
 
-export const HandshakeStoppedState = new State<HandshakeStoppedStateContext, Events, Effects>('STOPPED');
+export const HandshakeStoppedState = new State<HandshakeStoppedStateContext, Events, Effects>('HANDSHAKE_STOPPED');
 
-HandshakeStoppedState.on(subscriptionChange.type, (_context, event) =>
+HandshakeStoppedState.on(subscriptionChange.type, (context, event) =>
   HandshakeStoppedState.with({
     channels: event.payload.channels,
     groups: event.payload.groups,
+    cursor: context.cursor,
   }),
 );
 
-HandshakeStoppedState.on(reconnect.type, (context) => HandshakingState.with({ ...context }));
+HandshakeStoppedState.on(reconnect.type, (context, event) =>
+  HandshakingState.with({
+    ...context,
+    cursor: {
+      timetoken: event.payload?.timetoken ?? context.cursor?.timetoken ?? '0',
+      region: event.payload?.region ?? 0,
```

Comment:
Should we check `region` in `context` in the same way as with `timetoken` to make sure that we avoid any issues because of old `region`?

Reply:
updated: and defaulting to 0 only when event/context both doesn't have cursor info with region